### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This should be ready now :D

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,7 +72,8 @@
   "dependencies": {},
   "peerDependencies": {
     "react": ">=16.8.0",
-    "react-dom": ">=16.8.0"
+    "react-dom": "optional:>=16.8.0",
```
Comment (JoviDeCroock) [inline package.json]: This is tentative, let's see if this works

File: src/context/emitter.ts [hunk 2]
```diff
@@ -6,45 +8,48 @@ interface EmitMap {
...
-    fieldId.forEach((f) => { notify(`${f}`); });
-  } else {
-    notify(`${fieldId}`);
+  const visited: Array<string> = [];
+  if (!Array.isArray(fieldId)) {
+    fieldId = [fieldId];
   }
-  notify('all');
+
+  unstable_batchedUpdates(() => {
```
Comment (JoviDeCroock) [inline src/context/emitter.ts]: This should def lift us up to most performant form lib 🤞

File: src/context/emitter.ts [hunk 3]
```diff
@@ -6,45 +8,48 @@ interface EmitMap {
...
 }
 
 export function emit(fieldId: string | Array<string>) {
-  if (Array.isArray(fieldId)) {
-    fieldId.forEach((f) => { notify(`${f}`); });
-  } else {
-    notify(`${fieldId}`);
+  const visited: Array<string> = [];
+  if (!Array.isArray(fieldId)) {
+    fieldId = [fieldId];
```
Comment (JoviDeCroock) [inline src/context/emitter.ts]: This de-opts in the v8 engine but I'm okay with this small one

File: src/context/emitter.ts [hunk 4]
```diff
@@ -6,45 +8,48 @@ interface EmitMap {
...
+    });
+    notify('all');
+  });
 }
 
 function notify(fieldId: string) {
   if (mapping[fieldId]) {
-    mapping[fieldId].forEach((cb) => { cb(); });
+    // @ts-ignore
+    mapping[fieldId].some((cb) => { cb(); });
```
Comment (JoviDeCroock) [inline src/context/emitter.ts]: using .some to save some bytes (2KB) was my goal

File: src/helpers/batched.native.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+// @ts-ignore
```
Comment (JoviDeCroock) [inline src/helpers/batched.native.ts]: I've really started thinking that ts is biting me more as of late

File: src/helpers/useState.ts [hunk 6]
```diff
@@ -3,15 +3,17 @@ import { set } from './operations';
 
 type Output = [object, (id: string, value: any) => void, (newState: object) => void];
 
+export const EMPTY_ARRAY = [];
```
Comment (JoviDeCroock) [inline src/helpers/useState.ts]: Little byte saving trick for dep arrays for hooks

File: src/useFieldArray.ts [hunk 7]
```diff
@@ -55,10 +52,10 @@ export default function useFieldArray<T = any>(
         [value],
       ),
       remove: React.useCallback(
-        (element: T | number) => {
+        (index: number) => {
           ctx.setFieldValue(
             fieldId,
-            value.filter(x => x !== (typeof element === 'number' ? value[element] : element)),
+            value.filter((_, i) => i !== index),
```
Comment (JoviDeCroock) [inline src/useFieldArray.ts]: Deprecating removing through element since it doesn't make that much sense

Comment (Pruxis) [reply src/useFieldArray.ts]: Didn't even know this existed :p

**Pull request merged**